frontend/public/components/build.tsx

```diff
@@ -58,7 +60,15 @@ export const BuildsDetails: React.SFC<BuildsDetailsProps> = ({obj: build}) => {
   </div>;
 };
 
-const pages = [navFactory.details(BuildsDetails), navFactory.editYaml()];
+const envPath = ['spec', 'strategy', 'sourceStrategy'];
+const environmentComponent = (props) => <EnvironmentPage
+  obj={props.obj}
+  rawEnvData={props.obj.spec.strategy.sourceStrategy}
```

Comment:
This already merged, but I suspect this can cause runtime errors for other build strategies. There are a number of different strategies we need to handle: `dockerStrategy`, `sourceStrategy`, `customStrategy`, and `jenkinsPipelineStrategy` are all possible. Any of those can have env vars.

https://github.com/openshift/api/blob/master/build/v1/types.go#L555-L572

Reply:
Yup, I got this just now:

From a build config, clicking Environment:
![screen shot 2018-05-24 at 10 51 19 am](https://user-images.githubusercontent.com/280512/40493590-08e8c8de-5f41-11e8-97e4-529727895bdd.png)

Runtime error:
![screen shot 2018-05-24 at 10 51 11 am](https://user-images.githubusercontent.com/280512/40493589-08cb1ed8-5f41-11e8-8b74-63d0257a4d7d.png)

This is white-screen-of-death, have to refresh the app to recover.